ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainActivity.kt

```diff
@@ -31,6 +37,42 @@ class MainActivity : ComponentActivity() {
             mService = null
         }
     }
+    private fun copyFilesToExternalDir() {
+        val am = resources.assets
+        var files: Array<String>? = null
+
+        try {
+            files = am.list("models/")
+        } catch (e: java.lang.Exception) {
+            Log.e(TAG, "#### Failed to get asset file list")
+            e.printStackTrace()
+            return
+        }
+
+        // Copy files into app-specific directory.
+        for (filename in files!!) {
+            try {
+                val inFile = am.open("models/$filename")
+                val outDir = getExternalFilesDir(null)!!.absolutePath
+                val outFile = File(outDir, filename)
+                val out: OutputStream = FileOutputStream(outFile)
+
+                val buffer = ByteArray(1024)
+                var read: Int
+                while ((inFile.read(buffer).also { read = it }) != -1) {
+                    out.write(buffer, 0, read)
+                }
+
+                inFile.close()
+                out.flush()
+                out.close()
+            } catch (e: IOException) {
+                Log.e(TAG, "Failed to copy file: $filename")
+                e.printStackTrace()
+                return
+            }
+        }
+    }
```

Comment:
Looks OK to me. However, IMHO, we can implement only using Kotlin libraries without the JAVA libraries.
Please consider ```let```, ```run```, ```also``` next time.

Reply:
Now functions from old Java applications are used as is, with only the syntax changed.
It will be better to use Kotlin libraries.
I created an issue at AIP/mlops-on-android.
Thank you :)